_posts/2018-06-27-inception-pipelines-pt4.markdown

```diff
@@ -0,0 +1,171 @@
+
+The second, and subsequent, pipeline is for doing stuff within the account; they are an aspect of why the account exists in the first place. For example, if the account is for hosting a static website, this is where you put in all the CloudFormation code for configuring CloudFront, S3 buckets, Route 53 domain entries, etc. These items are more likely to have constant change and/or be more transient in lifespan. Another way to think about these aspects is that they are aspects that you want to delete (with the pipeline) but not delete the account.
+
+### What every organisation should have
+
+![aws accounts]({{ site.url }}/img/inception-pipelines/part-4-what-every-organisation-needs.png)
+
+#### Root AWS Account
+
+AWS accounts follow a tree like structure where an account can have zero to many child accounts. At the very top of this hierarchy is the root account. This account should be one of the most locked down accounts you have in the tree, both in terms of what people can do and the AWS services that are run inside it. Ideally in here there is only a single Inception Pipeline which is used to manage things which are global.
+
+A perfect use case here is defining IAM users that can assume roles in other accounts. This gives you the following benefits:
+
+1. Centralised management of IAM users. If someone starts or leaves your organisation, you can add or delete them from a single location rather than hunting through all your other AWS accounts.
+2. Centralised management of permissions. You define an IAM Group in the root account that can then assume a role in a child account. So, if the child account defines an Administrator role, you match that with a group here and then add/remove IAM users as required.
+3. All changes to your IAM users and groups are managed as code. This gives you a minimalistic audit trail of who made changes.
+
+Next month I'll expand upon this idea with code snippets. One other consideration too with only creating users in the Root is the [IAM Service Limits](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-limits.html#reference_iam-limits-entities). It's probably not an issue unless you literally have thousands of users, in which case you'll be better off using a Single Sign Identity Provider (because your root account now only contains roles).
+
+#### Audit AWS Account
+
+This account is where all things security and audit-ty related go. The account-level Inception Pipeline would define three IAM roles:
+
+* A heavily controlled **Administrator** account. Once defined, it would only need to be used in the case of emergency to perform actions (most likely in the AWS Console itself).
+* A more open role that can be assumed to make changes via the other pipelines. Called something like **SecurityOwner**?
+* Finally, a read-only role that can be given to developers, auditors, etc to view logs and other files stored in this account, and possibly called something like **SecurityReviewer**.
```

Comment:
remove the comma after account and change `and possibly` to `may be` or `could be`